src/plugins/ui_feature/find_and_replace.ts

```diff
@@ -44,26 +30,52 @@ interface SearchMatch {
 
 export class FindAndReplacePlugin extends UIPlugin {
   static layers = [LAYERS.Search];
-  static getters = ["getSearchMatches", "getCurrentSelectedMatchIndex"] as const;
-  private searchMatches: SearchMatch[] = [];
+  static getters = [
+    "getSearchMatches",
+    "getCurrentSelectedMatchIndex",
+    "getSearchOptions",
+    "getTotalMatchesAllSheets",
+    "getTotalMatchesActiveSheet",
+    "getTotalMatchesSpecificRange",
+  ] as const;
+
+  private allSheetsMatches: SearchMatch[] = [];
+  private activeSheetMatches: SearchMatch[] = [];
+  private specificRangeMatches: SearchMatch[] = [];
+
   private selectedMatchIndex: number | null = null;
   private currentSearchRegex: RegExp | null = null;
-  private searchOptions: SearchOptions = {
+  private searchOptions: SearchOptionsInternal = {
     matchCase: false,
     exactMatch: false,
     searchFormulas: false,
+    searchScope: "allSheets",
+    specificRange: undefined,
   };
   private toSearch: string = "";
   private isSearchDirty = false;
 
+  get searchMatches(): SearchMatch[] {
```

Comment:
Every other plugin starts with allowDispatch/handle (or range management), which I think is nice because that's the most important part of the plugin. But it's a detail